### PR TITLE
App scaffold

### PR DESCRIPTION
**Pull request by mateuszgorniak**
### Change
Create basic components to show (hardcoded for now) current and upcoming events.
https://monosnap.com/file/goVZfqFkPhnNsxvNLAZdjf0Dp3YMEf

### CODE REVIEW

File: src/components/BasicDataColumn.vue [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <current-event-details v-bind='currentEvent'></current-event-details>
+  </div>
+</template>
+
+<script>
+  import CurrentEventDetails from './CurrentEventDetails.vue'
+
+  export default {
+    components: {
+      'current-event-details': CurrentEventDetails
```
Comment (michalsnik) [inline src/components/BasicDataColumn.vue]: `CurrentEventDetails` is enough

File: src/components/BasicDataColumn.vue [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<template>
+  <div>
+    <h1>now</h1>
+    <current-event-details v-bind='currentEvent'></current-event-details>
```
Comment (michalsnik) [inline src/components/BasicDataColumn.vue]: Better to be explicit with props and use shortened syntax, plus tags can be self closed:
```
<current-event-details :name="currentEvent.name" :time="currentEvent.time" />
```

File: src/components/CurrentEventDetails.vue [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<template>
+  <div class='current-event'>
```
Comment (michalsnik) [inline src/components/CurrentEventDetails.vue]: You should use double quotes `"` instead

File: src/components/CurrentEventDetails.vue [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    <p class='current-event__name'>{{name}}</p>
+    <p class='current-event__time'>{{time}}</p>
+  </div>
+</template>
+
+<script>
+  export default {
+    props: ['name', 'time']
```
Comment (michalsnik) [inline src/components/CurrentEventDetails.vue]: It's good to take advantage of built in prop types:
```
props: {
  name: {
    type: String,
    required: true
  },
  time: {
    type: Date,
    required: true,
  }
}

File: src/components/CurrentEventDetails.vue [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  }
+</script>
+
+<style scoped>
```
Comment (michalsnik) [inline src/components/CurrentEventDetails.vue]: You can add `lang="scss"` and take advantage of SASS features, like e.g. nesting

File: src/components/CurrentTime.vue [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    },
+    methods: {
+      updateTime () {
+        window.setInterval(() => {
+          this.time = new Date().getTime()
+        }, 1000)
+      }
+    },
+    mounted () {
+      this.updateTime()
```
Comment (michalsnik) [inline src/components/CurrentTime.vue]: You should also clearInterval when the component is being destroyed.

File: src/components/DetailsColumn.vue [hunk 7]
```diff
@@ -0,0 +1,22 @@
+<template>
+  <div>
+    <current-time></current-time>
```
Comment (michalsnik) [inline src/components/DetailsColumn.vue]: `<current-time />`

File: src/components/DetailsColumn.vue [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  </div>
+</template>
+
+<script>
+  import CurrentTime from './CurrentTime.vue'
+  import EventsList from './EventsList.vue'
+
+  export default {
+    components: {
+      'current-time': CurrentTime,
```
Comment (michalsnik) [inline src/components/DetailsColumn.vue]:
```
{
  CurrentTime,
  EventsList,
}
```

File: src/components/EventsList.vue [hunk 9]
```diff
@@ -0,0 +1,24 @@
+<template>
+  <div>
+    <events-list-header></events-list-header>
```
Comment (michalsnik) [inline src/components/EventsList.vue]: `<events-list-header />`

File: src/components/EventsList.vue [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    <div v-for='event in nextEvents'>
```
Comment (michalsnik) [inline src/components/EventsList.vue]: This element should also have a `key`:
```
<div v-for="event in nextEvents" :key="event.id">
```

File: src/components/EventsList.vue [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+      <events-list-row v-bind='event'></events-list-row>
```
Comment (michalsnik) [inline src/components/EventsList.vue]: Pass event either explicitly:
```
<events-list-row :event="event" />
```
or using primitives:
```
<events-list-row :name="event.name" ... />
```

File: src/components/EventsList.vue [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+  </div>
+</template>
+
+<script>
+  import EventsListHeader from './EventsListHeader.vue'
+  import EventsListRow from './EventsListRow.vue'
+
+  export default {
+    components: {
+      'events-list-header': EventsListHeader,
```
Comment (michalsnik) [inline src/components/EventsList.vue]: Same as above

File: src/components/EventsListHeader.vue [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+  import * as moment from 'moment'
+
+  export default {
+    data () {
+      return {
+        date: new Date()
+      }
+    },
+    filters: {
+      formattedDate (value) {
```
Comment (michalsnik) [inline src/components/EventsListHeader.vue]: Isn't this duplicated? If yes you should consider creating separate folder with global filters, and set them globally in `main.js`

File: src/components/EventsListHeader.vue [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+      }
+    },
+    methods: {
+      updateDate () {
+        window.setInterval(() => {
+          this.date = new Date()
+        }, 1000)
+      }
+    },
+    mounted () {
```
Comment (michalsnik) [inline src/components/EventsListHeader.vue]: You forgot to clearInterval

File: src/components/EventsListHeader.vue [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+    display: flex;
+    flex-direction: row;
+    font-size: 16px;
+    justify-content: space-between;
+    margin: 0 35px 0 20px;
+  }
+
+  .current-date__label {
+    font-weight: 400;
+    float: left;
```
Comment (michalsnik) [inline src/components/EventsListHeader.vue]: Don't use floats when you use flexbox. `space-between` on parent element should do the trick just fine.

File: src/components/Main.vue [hunk 16]
```diff
@@ -1,19 +1,50 @@
...
 </template>
 
 <script>
-  export default {}
+  import BasicDataColumn from './BasicDataColumn.vue'
+  import DetailsColumn from './DetailsColumn.vue'
+
+  export default {
+    components: {
+      'basic-data-column': BasicDataColumn,
```
Comment (michalsnik) [inline src/components/Main.vue]: Same

File: src/components/EventsListHeader.vue (hunk 13 above)
Comment (mateuszgorniak) [reply src/components/EventsListHeader.vue]: It's not the same, but in the future it might be quite nice idea to store all filters in the one place 👍

**Pull request merged**